pom.xml

```diff
@@ -175,6 +172,7 @@
                         <artifactId>logback-classic</artifactId>
                     </exclusion>
                 </exclusions>
+                <scope>provided</scope>
```

Comment:
Similar to `kafka_2.13` should we mark zookeeper as `optional`. I think the case here is even stronger.

Reply:
Excellent point.   I needed to read up.  I found https://medium.com/@danismaz.furkan/difference-between-optional-true-optional-and-scope-provided-scope-7404ec24fb59 helpful.

I think you are right, zookeeper should be `optional`.  kafka should remain as `provided` as we currently need it on all code paths.  The user must provide it.

Not for now - it occurs to me that the junit ext ought to be applying the service loader pattern to KafkaCluster with container and in-vm implementations contributed from separate modules.  The common code that relies on kafka server classes should looked at with a view to eliminating the dependency (mostly config constants IIRC).  That way, if a user wanted to exclusively use the container-based implementation, they could avoid the kafka-server dependencies completely.   Like I say, I don't see any pressing need to do this now, but wanted to share the idea.

Reply:
My thinking (although unvalidated) was that Kafka was optional as well if users are only using the in containers provider.

Though I could easily believe we actually need it regardless so provided is the right choice.

Reply:
It turns out `optional` flag is ignored by dependencyManagement (MNG-5227).
I'll leave the tag as it communicates intent but re-added `provided` scope.